upup/pkg/fi/cloudup/template_functions_test.go

```diff
@@ -0,0 +1,290 @@
+						LogLevel:      3,
+					},
+				},
+			},
+			expectedArgv: []string{
+				"--v=3",
+				"--cloud-provider=openstack",
+				"--use-service-account-credentials=true",
+			},
+		},
+		{
+			desc: "No CloudProvider Configuration",
+			cluster: &kops.Cluster{
+				Spec: kops.ClusterSpec{
+					ExternalCloudControllerManager: &kops.CloudControllerManagerConfig{
+						LogLevel: 3,
+					},
+				},
+			},
+			expectedError: fmt.Errorf("Cloud Provider is not set"),
+		},
+		{
+			desc: "k8s cluster name",
+			cluster: &kops.Cluster{Spec: kops.ClusterSpec{
+				CloudProvider: string(kops.CloudProviderOpenstack),
+				ExternalCloudControllerManager: &kops.CloudControllerManagerConfig{
+					ClusterName: "k8s",
+				},
+			}},
+			expectedArgv: []string{
+				"--v=2",
+				"--cloud-provider=openstack",
+				"--cluster-name=k8s",
+				"--use-service-account-credentials=true",
+			},
+		},
+		{
+			desc: "Default Configuration",
+			cluster: &kops.Cluster{Spec: kops.ClusterSpec{
+				CloudProvider: string(kops.CloudProviderOpenstack),
+				ExternalCloudControllerManager: &kops.CloudControllerManagerConfig{
+					Master: "127.0.0.1",
+				},
+			}},
+			expectedArgv: []string{
+				"--master=127.0.0.1",
+				"--v=2",
+				"--cloud-provider=openstack",
+				"--use-service-account-credentials=true",
+			},
+		},
+		{
+			desc: "Cluster-cidr Configuration",
+			cluster: &kops.Cluster{Spec: kops.ClusterSpec{
+				CloudProvider: string(kops.CloudProviderOpenstack),
+				ExternalCloudControllerManager: &kops.CloudControllerManagerConfig{
+					ClusterCIDR: "10.0.0.0/24",
+				},
+			}},
+			expectedArgv: []string{
+				"--v=2",
+				"--cloud-provider=openstack",
+				"--cluster-cidr=10.0.0.0/24",
+				"--use-service-account-credentials=true",
+			},
+		},
+		{
+			desc: "AllocateNodeCIDRs Configuration",
+			cluster: &kops.Cluster{Spec: kops.ClusterSpec{
+				CloudProvider: string(kops.CloudProviderOpenstack),
+				ExternalCloudControllerManager: &kops.CloudControllerManagerConfig{
+					AllocateNodeCIDRs: fi.Bool(true),
+				},
+			}},
+			expectedArgv: []string{
+				"--v=2",
+				"--cloud-provider=openstack",
+				"--allocate-node-cidrs=true",
+				"--use-service-account-credentials=true",
+			},
+		},
+		{
+			desc: "ConfigureCloudRoutes Configuration",
+			cluster: &kops.Cluster{Spec: kops.ClusterSpec{
+				CloudProvider: string(kops.CloudProviderOpenstack),
+				ExternalCloudControllerManager: &kops.CloudControllerManagerConfig{
+					ConfigureCloudRoutes: fi.Bool(true),
+				},
+			}},
+			expectedArgv: []string{
+				"--v=2",
+				"--cloud-provider=openstack",
+				"--configure-cloud-routes=true",
+				"--use-service-account-credentials=true",
+			},
+		},
+		{
+			desc: "CIDRAllocatorType Configuration",
+			cluster: &kops.Cluster{Spec: kops.ClusterSpec{
+				CloudProvider: string(kops.CloudProviderOpenstack),
+				ExternalCloudControllerManager: &kops.CloudControllerManagerConfig{
+					CIDRAllocatorType: fi.String(string(ipam.RangeAllocatorType)),
```

Comment:
It seems like all of the new vendored code is only to use this one constant, is that correct? I know we've been trying to become less dependent on `kubernetes/kubernetes`, so rather than adding all of the vendored code could we just reuse the string `"RangeAllocator"` here?